src/Surfnet/StepupMiddleware/ApiBundle/Controller/DeprovisionController.php

```diff
@@ -30,6 +36,9 @@ class DeprovisionController extends AbstractController
     public function __construct(
         private readonly DeprovisionServiceInterface $deprovisionService,
         private readonly UserDataFormatterInterface $formatHelper,
+        private readonly IdentityRepository $identityRepository,
+        private readonly SraaRepository $sraaRepository,
+        private readonly RaListingRepository $raListingRepository,
```

Comment:
If you allow me to be critital (I'll assume yes :stuck_out_tongue:):
Should the dry run be handled by a service aswel?
Three new services injected smells like the controller does more than it should.

Reply:
Moved the logic to the deprovision service itself.  I didn't want to move it to a separate service because the dry-run should also handle the deprovisioning, only without the call to the deprovision method on the aggregate. So when changes are made on the deprovisioning this should also be dealt with for the dry-run. And by doing this in the same service one should be aware of this.